### ISSUE TITLE
Ruby version from Gemfile is ignored on deploy

### ISSUE THREAD

**Issue by lenn4rd** [flynn/flynn#3319]
I'm trying to deploy a Rails app pinned to Ruby 2.3.1 in the Gemfile but deployment fails when bundling the gems. This looks similar to #2929.

Just to rule out some misconfiguration on my side I took the `ruby-flynn-example` app and tried both Ruby 2.2.4 and 2.3.1. With 2.2.4 it deploys successfully (after changing Event Machine version to 1.0.4 which is a requirement for that Ruby version).

However after setting Ruby to 2.3.1 in the Gemfile I get this output:

```
$ git branch
* master

$ git push flynn master
Counting objects: 3, done.
Delta compression using up to 8 threads.
Compressing objects: 100% (3/3), done.
Writing objects: 100% (3/3), 342 bytes | 0 bytes/s, done.
Total 3 (delta 1), reused 0 (delta 0)
-----> Building ruby-example...
-----> Ruby app detected
-----> Compiling Ruby/Rack
-----> Using Ruby version: ruby-2.2.4
-----> Installing dependencies using bundler 1.12.5
       Running: bundle install --without development:test --path vendor/bundle --binstubs vendor/bundle/bin -j4 --deployment
       Your Ruby version is 2.2.4, but your Gemfile specified 2.3.1
       Bundler Output: Your Ruby version is 2.2.4, but your Gemfile specified 2.3.1
       !
       !     Failed to install gems via Bundler.
       !
remote: ERROR: Build failed: exec: job exited with status 1
To https://git.gzwx.flynnhub.com/ruby-example.git
 ! [remote rejected] master -> master (pre-receive hook declined)
error: failed to push some refs to 'https://git.gzwx.flynnhub.com/ruby-example.git'
```

I created a [gist with my changes](https://gist.github.com/lenn4rd/f56052fa511417611021b14c063ce33a). Let me know if you need any additional information, I'm happy to help.

**Comment by lmars** [comment]
@lenn4rd it looks like the version in your `Gemfile.lock` (`2.2.4p230`) does not match the version in your `Gemfile` (`2.3.1`), are you using `2.3.1` locally?

**Comment by lenn4rd** [comment]
You're right, I was. I'm using rbenv and first tried 2.2.4 which got added to the `Gemfile.lock`. When I updated the Ruby version in the `Gemfile` Bundler didn't update it apparently. I knew that the Bundler version is stored, but never noticed the Ruby version is, too. Maybe this changed recently.

This seems to have been a bug with Bundler that has been fixed and, judging from the comments in https://github.com/bundler/bundler/issues/4595, will be included in Bundler 1.12.6.
